lib/rules/named-import.js

```diff
@@ -0,0 +1,238 @@
+/**
+ * @fileoverview Named imports on React
+ * @author Angelo Abdoelsamath
+ */
+
+'use strict';
+
+const docsUrl = require('../util/docsUrl');
+
+const DEFAULT_TYPE = 'import';
+
+module.exports = {
+  meta: {
+    fixable: 'code',
+    docs: {
+      description: 'Enforce or prohibit named imports',
+      category: 'Best Practices',
+      recommended: false,
+      url: docsUrl('named-import')
+    },
+    schema: [
+      {
+        type: 'string',
+        enum: ['property', 'import']
+      },
+      {
+        type: 'object',
+        patternProperties: {
+          '.': {
+            type: 'string',
+            enum: ['property', 'import']
+          }
+        },
+        additionalProperties: false
+      }
+
+    ],
+    messages: {
+      usePropertyAccessor: 'use React.{{name}}',
+      useNamedImport: 'Import {{name}} from React',
+      fixImportStatement: 'Fix import statement'
+    }
+  },
+  /**
+   *
+   * @param {RuleContext} context
+   * @returns {any}
+   */
+  create(context) {
+    // ignore non-modules
+    if (context.parserOptions.sourceType !== 'module') {
+      return {};
+    }
+    const options = context.options;
+    const mode = options[0] || DEFAULT_TYPE;
+    const overrides = options[1] || {};
+
+    let propertySpecifiers = [];
+    let specifiers = [];
+
+    let importDeclaration;
+
+    /** @type {Context} */
+    const sourceCode = context.getSourceCode();
+
+    function isNameOfType(name, type) {
+      return (overrides[name] || mode) === type;
+    }
+
+    function getImportDeclarationString() {
+      let str = 'React';
```

Comment:
we probably need a way to configure whether the user is using the jsx runtime or not.

i think we'll need to add a global plugin setting, that this rule (and a few others) can look to. When set, we'd only provide the `React` import if something referenced it, not automatically.

Reply:
Can you point me in the right direction to make this work. I'm not sure how to implement this

Reply:
By global settings, I mean the "settings" stuff referenced here: https://github.com/yannickcr/eslint-plugin-react#configuration

If we add a boolean `jsx-runtime` setting, when it's false, they're using the React 16 jsx compilation, and React must always be in scope. When it's true, they're using the React 17 jsx compilation, and React need not be in scope.